android/java/org/chromium/chrome/browser/app/domain/PortfolioModel.java

```diff
@@ -133,11 +142,63 @@ public void discoverAssetsOnAllSupportedChains() {
+            BraveWalletBaseActivity braveWalletBaseActivity,
+            Callbacks.Callback1<PortfolioHelper> callback) {
+        synchronized (mLock) {
+            clear();
+            if (mJsonRpcService == null) {
+                return;
+            }
+            NetworkModel.getAllNetworks(
+                    mJsonRpcService, mSharedData.getSupportedCryptoCoins(), allNetworks -> {
+                        mAllNetworkInfos = new ArrayList<>(allNetworks);
+                        if (selectedNetwork.chainId.equals(
+                                    NetworkUtils.getAllNetworkOption(mContext).chainId)) {
+                            mKeyringService.getKeyringsInfo(
+                                    mSharedData.getEnabledKeyrings(), keyringInfos -> {
+                                        AccountInfo[] accountInfos =
+                                                WalletUtils
+                                                        .getAccountInfosFromKeyrings(keyringInfos)
+                                                        .toArray(new AccountInfo[0]);
+                                        mPortfolioHelper =
+                                                new PortfolioHelper(braveWalletBaseActivity,
+                                                        mAllNetworkInfos, accountInfos);
+                                        mPortfolioHelper.setSelectedNetworks(
+                                                NetworkUtils.nonTestNetwork(mAllNetworkInfos));
+                                        fetchUserAssetsAndDetails(mPortfolioHelper, callback);
+                                    });
+                        } else {
+                            mKeyringService.getKeyringInfo(
+                                    AssetUtils.getKeyringForCoinType(selectedNetwork.coin),
+                                    keyringInfo -> {
+                                        mPortfolioHelper =
+                                                new PortfolioHelper(braveWalletBaseActivity,
+                                                        mAllNetworkInfos, keyringInfo.accountInfos);
+                                        mPortfolioHelper.setSelectedNetworks(
+                                                Arrays.asList(selectedNetwork));
+                                        fetchUserAssetsAndDetails(mPortfolioHelper, callback);
+                                    });
+                        }
+                    });
+        }
+    }
+
+    @Override
+    public void onDiscoverAssetsCompleted(BlockchainToken[] discoveredAssets) {
+        _mIsDiscoveringUserAssets.postValue(false);
+    }
+
     // Clear state
     public void clear() {
         _mNftModels.postValue(Collections.emptyList());
     }
 
+    private void fetchUserAssetsAndDetails(
+            PortfolioHelper portfolioHelper, Callbacks.Callback1<PortfolioHelper> callback) {
+        portfolioHelper.fetchAllAssetsAndDetails(callback);
+    }
+
```

Comment:
Considering this method is just one single line, I would remove this method entirely and simply replace its occurrences with `portfolioHelper.fetchAllAssetsAndDetails(callback);`.

Reply:
Don't think it is required, no harm in using same method at two places, even for small things.
I would strongly suggest to add `nit` for subjective comments to focus on primary concerns first and to avoid blocking PR merging.

Reply:
> I would strongly suggest to add nit for subjective comments

I do add `nit` when it's the right case.

I'm considering this method as something that does not add any tangible value I'm afraid :-(

It's also considered an antipattern (AHA), it hides the real implementation, it does not avoid any code duplication as it's a single line, and even the name does not add any benefits as `fetchUserAssetsAndDetails` calls `PortfolioHelper#fetchAllAssetsAndDetails`.

Just to be very transparent: I'm not against one-liner methods. it's okay to use them when they add value. But here it's just a proxy method calling a method of `PortfolioHelper` with the exact same name.

```java
void doSomething() {
    portfolioHelper.doSomething();
}
```

Removing this method makes the code easier to follow, easier to maintain, and the entire class slightly smaller. I don't see any valid reason that should lead us to accept such an antipattern. 

Further links to delve deeper into the topic:

* https://mytwoocents.com/2017/03/25/one-line-methods-can-easily-become-an-antipattern/
* https://en.wikipedia.org/wiki/Don%27t_repeat_yourself (AHA principle)

Reply:
I think we can both find pros and cons of each side. The idea here is to construct the `portfolioHelper` and perform the fetching in a separate method.
Won't consider this as an ideal candidate of AHA or one-liner because the idea here is to separate construction and calling. Additionally, it has been simplified to use the same method instead of calling `fetchUserAssetsAndDetails` and `calculateBalances` for single and multiple network value.

Here's some more details on `Nits`: https://google.github.io/eng-practices/review/reviewer/standard.html#:~:text=Reviewers%20should%20always%20feel%20free,they%20could%20choose%20to%20ignore.

> Reviewers should always feel free to leave comments expressing that something could be better, but if it’s not very important, prefix it with something like “Nit: “ to let the author know that it’s just a point of polish that they could choose to ignore.

Again, many things can be subjective here.

Reply:
I never mentioned that we should do this always `construct something in one place and perform some one line action in a separate method` nor that it is a bad practice, it depends. 
This is just to separate the flow, maybe renaming the method would have been more clearer or a better choice. But, if the preference is to move the call along with the construction of object then will move.

Reply:
I agree that having a method called `fetchUserAssetsAndDetails` just call `fetchAllAssetsAndDetails` is very confusing. Pick one or the other, whichever makes more sense here